core/src/gl/texture.cpp

```diff
@@ -64,23 +64,27 @@ void Texture::setData(const GLuint* _data, unsigned int _dataSize) {
     m_dirty = true;
 }
 
-void Texture::setSubData(const GLuint* _subData, unsigned int _xoff, unsigned int _yoff, unsigned int _width,
-                         unsigned int _height) {
+void Texture::setSubData(const GLuint* _subData, unsigned int _xoff, unsigned int _yoff,
+                         unsigned int _width, unsigned int _height) {
 
-    std::unique_ptr<std::vector<GLuint>> subData(new std::vector<GLuint>(_subData, _subData + (_width * _height)));
-
-    // update m_data with subdata
     size_t bpp = bytesPerPixel();
     size_t divisor = sizeof(GLuint) / bpp;
-    for (size_t j = 0; j < _height; j++) {
-        size_t dpos = ((j + _yoff) * m_width + _xoff) / divisor;
-        size_t spos = (j * _width) / divisor;
-        std::memcpy(&m_data[dpos], &_subData[spos], _width * bpp);
+
+    // Init m_data if update() was not called after resize()
+    if (m_data.size() != (m_width * m_height) / divisor) {
+        m_data.resize((m_width * m_height) / divisor);
```

Comment:
ok, the 'fix: initialization of Texture m_data' is also necessary, but I have never hit this case yet :)